CMakeLists.txt

```diff
@@ -149,6 +149,7 @@ add_library(uamqp
     ${uamqp_h_files}
     ${socketlistener_c_files}
     )
+setTargetBuildProperties(uamqp)
```

Comment:
>setTargetBuildProperties [](start = 0, length = 24)

If a warning is spewed by the compiler, then we should fix the warnings, rather than suppress them. Suppressing should be the last resort, only for the (unlikely) case that the compiler is wrong.

Reply:
This change it the **opposite** of suppression; it makes the compiler _more_ strict, not less.

Reply:
Oh, sorry, my bad :-). By all means!

---
In reply to: [174605521](https://github.com/Azure/azure-uamqp-c/pull/224#discussion_r174605521) [](ancestors = 174605521)

Reply:
I started the build to merge it in.

Reply:
Thanks!